app/lib/cognito.server.ts

```diff
@@ -58,6 +69,59 @@ export function extractAttributeRequired(
   return value
 }
 
+/**
+ * Gets another user from cognito
+ *
+ * @param sub - the sub of another user
+ * @returns a user if found, otherwise undefined
+ */
+export async function getCognitoUserFromSub(sub: string) {
+  const escapedSub = sub.replaceAll('"', '\\"')
+  const user = (
+    await cognito.send(
+      new ListUsersCommand({
+        UserPoolId,
+        Filter: `sub = "${escapedSub}"`,
+      })
+    )
+  )?.Users?.[0]
+
+  if (!user?.Username)
+    throw new Response('Requested user does not exist', {
+      status: 400,
+    })
+
+  return user
+}
+
+export async function listUsers() {
+  const pages = paginateListUsers(
+    { client: cognito },
+    {
+      UserPoolId,
+    }
+  )
+  const users: UserData[] = []
+  for await (const page of pages) {
+    const nextUsers = page.Users
+    if (nextUsers)
+      users.push(
+        ...nextUsers
+          .filter((user) => Boolean(extractAttribute(user.Attributes, 'email')))
+          .map((user) => ({
+            sub: extractAttributeRequired(user.Attributes, 'sub'),
+            email: extractAttributeRequired(user.Attributes, 'email'),
+            name: extractAttribute(user.Attributes, 'name'),
+            affiliation: extractAttribute(
+              user.Attributes,
+              'custom:affiliation'
+            ),
+          }))
+      )
+  }
+  return users
+}
```

Comment:
This function may take a painfully long time to execute in production with thousands of users. Instead of displaying a list of all of the users, could we have an autocomplete search box like we do for peer endorsements?

Reply:
Yea, I will implement that in the next commit